### PR TITLE
Rufus hamade ui improvements

### PR DESCRIPTION
**Pull request by RufusHamade**
A collection of UI bugfixes and improvements from the NYT work.

Not sure if you are doing reviews any more.  If not, let me know who should be doing them now.

### CODE REVIEW

File: packages/frontend-web/src/app/platform/websocketService.ts [hunk 1]
```diff
@@ -81,6 +81,18 @@ export interface IPerUserData {
 //       When this is availabe, replace the "any" types in the code below.
```
Comment (iislucas) [inline packages/frontend-web/src/app/platform/websocketService.ts]: I think this comment is about the function below fixObject ?

File: packages/frontend-web/src/app/platform/websocketService.ts [hunk 2]
```diff
@@ -81,6 +81,18 @@ export interface IPerUserData {
...
 // TODO: API sending number IDs, but we expect strings due to the way the old REST code works.
 //       Convert for now.  But at some point need to refactor to use numbers.
+
+function fixObject(o: any) {
```
Comment (iislucas) [inline packages/frontend-web/src/app/platform/websocketService.ts]: Seems pretty clear that `o` is not any, but `{ categoryId: string | Stringable; id: string | Stringable, tagId: string |Stringable }`. where Stringable is something like `{ toString() : string }`

When you introduce new (especially small_) functions, much better to give them types.

**Comment by RufusHamade** [comment]
Hi Lucas,

Sorry, I'll try harder to break things up better.  In future, feel free to reject the review and ask for it broken into smaller chunks.  I'll not mind.

Looking at your review comments and disucssing things with NYT,  I think it is better to do the conversion to strings in the server rather than the client.  The NYT backend uses their article ID which is a text string.  And allowing strings would allow us to add a bit of redundancy into the IDs.  So, assuming you don't mind, I'll implement that.  It'll make your review suggestions moot as the relevant code would be deleted.

**Comment by iislucas** [comment]
cool, sounds good to me!

**Pull request merged**